src/MarkdownTextInput.web.tsx

```diff
@@ -559,7 +559,9 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
         }
 
         const text = processedValue !== undefined ? processedValue : '';
-        parseText(divRef.current, text, processedMarkdownStyle, text.length);
+
+        const currentCursorPostion = contentSelection?.current && contentSelection?.current.end <= text.length ? contentSelection.current.end : text.length;
+        parseText(divRef.current, text, processedMarkdownStyle, currentCursorPostion);
```

Comment:
How about moving this code to `parseUtils > parseText` function. For example we can add following code at the beginning of the function:
```ts
  const targetElement = target;
  let cursorPosition: number | null = curosrPositionIndex;
  if (cursorPosition && cursorPosition > text.length) {
    cursorPosition = text.length;
  }
```